src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -685,6 +685,14 @@ private async Task<ConnectBlocksResult> ConnectChainAsync(ChainedHeader newTip,
             {
                 using (this.performanceCounter.MeasureBlockConnectionFV())
                 {
+                    if (blockToConnect.ChainedHeader.Block == null)
+                    {
+                        // In some rare cases block can be null.
+                        // This can happen after a reorg failed and
+                        // we are trying to reconnect the old chain
+                        blockToConnect.ChainedHeader.Block = blockToConnect.Block;
```

Comment:
i am now unconvinced this is good, can we make call tomorrow with ivan?

Reply:
Ok but is it critical or perhaps we push and make the fix in another PR, because right now node is unstable and this fixes it.

Reply:
i understand, but i think we need to fix correctly

Reply:
Lets discuss tomorrow.